### PR TITLE
MA-4034/PreConfig_Bug

### PR DESCRIPTION
**Pull request by priyanka969**
Purpose : 
Raised this PR against the Bug reported for validation error in Lambda function name for Preconfig step.

Changes :

Following are major changes :

Modified template.yaml file for removing space in function name
Re-added changes in input json file for state machine.

### CODE REVIEW

File: src/json/state_machine_input.json [hunk 1]
```diff
@@ -119,6 +119,27 @@
...
+       "roles":[  
+
+       ],
+       "authorizer_user_guid":null,
+       "durationSeconds":0
+    },
+    "manageVpcPreConfig":{  
+       "account":null,
+       "region":null,
+       "regions":[
```
Comment (SonalOjha) [inline src/json/state_machine_input.json]: If tomorrow we have one more region will that automatically get addressed in the ManagedVPC preconfig statemachine

Comment (mayankbharadwaj) [reply src/json/state_machine_input.json]: Yes sonal it will be addressed.

File: src/json/state_machine_input.json [hunk 2]
```diff
@@ -119,6 +119,27 @@
...
+  "vpcpreconfig":{  
+    "credentials":null,
+    "federation":{  
+       "roles":[  
+
+       ],
+       "authorizer_user_guid":null,
+       "durationSeconds":0
+    },
+    "manageVpcPreConfig":{
```
Comment (RajkumarShivage) [inline src/json/state_machine_input.json]: Can we change it to "managedVpcPreConfig"

File: src/json/state_machine_input.json [hunk 3]
```diff
@@ -119,6 +119,27 @@
       ]
     }
   },
+  "vpcpreconfig":{
```
Comment (RajkumarShivage) [inline src/json/state_machine_input.json]: What is the difference between manageVpcPreConfig and vpcpreconfig?

File: src/json/state_machine_input.json (hunk 2 above)
Comment (mayankbharadwaj) [reply src/json/state_machine_input.json]: I analyzed the impact of this change in state machine and i have to make changes in 16 places in different files where i am using this reference object.

File: src/json/state_machine_input.json (hunk 3 above)
Comment (priyanka969) [reply src/json/state_machine_input.json]: vpcpreconfig is like parent JSON element and manageVpcPreConfig is the child JSON element.

**Pull request merged**